### PR TITLE
feat: make `MAX_REQUEST_BLOB_SIDECARS` and `MAX_BLOBS_PER_BLOCK` configurable

### LINKED ISSUE #7172: Make `MAX_REQUEST_BLOB_SIDECARS` and `MAX_BLOBS_PER_BLOCK` configurable
### Problem description

In ethereum/consensus-specs#3817, `MAX_BLOBS_PER_BLOCK` is moved from preset to config. 

This poses a challenge in Lodestar since [`BlobSidecarsByRootRequest`](https://github.com/ChainSafe/lodestar/blob/de0d6ab89b575ad20c593acf20d7c97d2c884ab1/packages/types/src/deneb/sszTypes.ts#L65) - a fixed length ssz container depends on it but `types` package has no access to `chain.config`.
### 

### Solution description

I believe this is the only place in the spec has a ssz container for which its definition relies on a variable from config.

We can either make a one time exception and define `BlobSidecarsByRootRequest` ssz in `beacon-node` or further refactor to be more future proof if such scenario happens again.

cc. @g11tech 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ensi321**
We need to make these two parameters configurable before we implement EIP-7691

- Any function that needs access to `MAX_BLOBS_PER_BLOCK` will need to have `config` passed in
- Move `BlobSidecarsByRootRequest` out of `packages/types` to `packages/beacon-node` since it is now "dynamic' depending on `config.MAX_REQUEST_BLOB_SIDECARS`.
- Any method with ssz response type `BlobSidecarsByRootRequest` will need to access `config`.

Closes #7172

### CODE REVIEW

File: packages/params/src/index.ts [hunk 1]
```diff
@@ -198,7 +197,6 @@ export const SYNC_COMMITTEE_SUBNET_SIZE = Math.floor(SYNC_COMMITTEE_SIZE / SYNC_
 
 export const MAX_REQUEST_BLOCKS = 2 ** 10; // 1024
 export const MAX_REQUEST_BLOCKS_DENEB = 2 ** 7; // 128
```
Comment (nflaig) [inline packages/params/src/index.ts:199]: as per spec those should be part of config as well, see [configs/mainnet.yaml#L146](https://github.com/ethereum/consensus-specs/blob/18af2af7e98d5cf6ac40145086622b4590447e9b/configs/mainnet.yaml#L146)

File: packages/beacon-node/src/util/types.ts [hunk 2]
```diff
@@ -12,3 +13,7 @@ export const signedBLSToExecutionChangeVersionedType = new ContainerType(
   {jsonCase: "eth2", typeName: "SignedBLSToExecutionChangeVersionedType"}
 );
 export type SignedBLSToExecutionChangeVersioned = ValueOf<typeof signedBLSToExecutionChangeVersionedType>;
+
+export const BlobSidecarsByRootRequestType = (config: Pick<ChainConfig, "MAX_REQUEST_BLOB_SIDECARS">) =>
```
Comment (nflaig) [inline packages/beacon-node/src/util/types.ts]: I don't think it's helpful to pick just the specific key, this is actually more error prone than making sure the proper config object is passed

File: packages/beacon-node/src/network/gossip/topic.ts [hunk 3]
```diff
@@ -213,7 +213,7 @@ export function getCoreTopicsAtFork(
 
   // After Deneb also track blob_sidecar_{index}
   if (ForkSeq[fork] >= ForkSeq.deneb) {
-    for (let index = 0; index < MAX_BLOBS_PER_BLOCK; index++) {
+    for (let index = 0; index < config.MAX_BLOBS_PER_BLOCK; index++) {
```
Comment (nflaig) [inline packages/beacon-node/src/network/gossip/topic.ts:216]: just a side note, we should use `BLOB_SIDECAR_SUBNET_COUNT` here instead, currently this does not matter as `BLOB_SIDECAR_SUBNET_COUNT == MAX_BLOBS_PER_BLOCK` but would be good to clean up some open todos left over from Deneb

Will put up a separate PR for that

File: packages/types/src/deneb/sszTypes.ts [hunk 4]
```diff
@@ -62,8 +61,6 @@ export const BlobIdentifier = new ContainerType(
   {typeName: "BlobIdentifier", jsonCase: "eth2"}
 );
 
-export const BlobSidecarsByRootRequest = new ListCompositeType(BlobIdentifier, MAX_REQUEST_BLOB_SIDECARS);
```
Comment (nflaig) [inline packages/types/src/deneb/sszTypes.ts:65]: moving this out of types should be fine as it's not a ssz container defined in the spec, only [specs/deneb/p2p-interface.md?plain=1#L284](https://github.com/ethereum/consensus-specs/blob/18af2af7e98d5cf6ac40145086622b4590447e9b/specs/deneb/p2p-interface.md?plain=1#L284)

**Review by nflaig — APPROVED** [review]
LGTM - there are a few more constants we should move into config but can be addressed separately

File: packages/validator/src/util/params.ts [hunk 5]
```diff
@@ -216,9 +216,10 @@ function getSpecCriticalParams(localConfig: ChainConfig): Record<keyof ConfigWit
     // # DenebPreset
     /////////////////
     FIELD_ELEMENTS_PER_BLOB: denebForkRelevant,
-    MAX_BLOBS_PER_BLOCK: denebForkRelevant,
     MAX_BLOB_COMMITMENTS_PER_BLOCK: denebForkRelevant,
     KZG_COMMITMENT_INCLUSION_PROOF_DEPTH: denebForkRelevant,
+    MAX_BLOBS_PER_BLOCK: denebForkRelevant,
+    MAX_REQUEST_BLOB_SIDECARS: denebForkRelevant,
```
Comment (nflaig) [inline packages/validator/src/util/params.ts]: `MAX_REQUEST_BLOB_SIDECARS` is part of network config, I don't think we need to check this from the vc side

We might rather wanna add it here
https://github.com/ChainSafe/lodestar/blob/99794d3b261bb50f9a6c7d9f6ca19291b3f30911/packages/validator/src/util/params.ts#L137-L138

**Review by nflaig — CHANGES_REQUESTED** [review]

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.24.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.24.0) :tada:

**Pull request merged**